Refactored API Routes into Separate Files

Kavipatel0 opened the pull request:
## Description
Moved API Routes from `index.ts` into a routes folder. Routes folder includes user and auth folders with respective route handlers. Routes are contained within the `user` and `auth` folders which each have their own respective route collector in `auth/index.ts` and `user/index.ts`. This funnels into a central route handler in `routes/index.ts` which is then `app.use()` in the original server `src/index.ts`.


Closes #240

AlexanderWangY commented on server/src/routes/index.ts:
```diff
@@ -0,0 +1,10 @@
+import authRouter from "./auth";
+import userRouter from "./user";
+
+const userAndAuthRouter = Router();
```
Change this name incase we have more than just user and auth. Make it applicable to any routes created.

AlexanderWangY commented on server/src/routes/user/getNearby.ts:
Rename this to something other than getNearby.ts this does not fit with our current naming conventions (look at the other files names in this folder)

AlexanderWangY commented on server/src/routes/auth/email.ts:
```diff
@@ -0,0 +1,39 @@
+import { Router } from "express";
+import Mailgun from "mailgun.js";
+
+const authUserEmailRoute = Router();
```
User? Isn't that another folder. Lets just keep this to authEmailRoute()

AlexanderWangY commented on server/src/index.ts:
```diff
@@ -156,166 +157,16 @@ app.get("/", (req, res) => {
-});
 
 // Error handling
 app.get("*", (req, res) => {
```
Maybe move these into a folder called `404` or `error`

AlexanderWangY commented on server/src/index.ts:
```diff
@@ -156,166 +157,16 @@ app.get("/", (req, res) => {
-    res.json(`Operation <GET /users${query}> failed.`);
-  }
-});
+app.use(userAndAuthRouter);
```
Again right here, change the name, this is really confusing for people if there are different routes other than Auth and User

AlexanderWangY reviewed (changes requested):
Make these changes and then it LGTM 🚀

h1divp commented:
I would rename the `index.ts` files into `routehandler.ts` or something a bit more clear in order to avoid confusion with `/server`'s entry point. Besides that, as long as this is tested, I think this is a great change. Perhaps something could be done with the socket endpoints as well.

Kavipatel0 commented:
👍🏽 
I implemented the suggested changes. Instead of having `index.ts` files managing the route handlers, I renamed them to be more descriptive. Now, `userRouteHandler.ts` handles the userRouter and `authRouteHandler.ts` manages the authRouter. Both of these are in `mainRouteHandler.ts` to improve clarity and structure.
😄

AlexanderWangY reviewed (approved):
LGTM 🚀

h1divp commented:
Thank you both !!
